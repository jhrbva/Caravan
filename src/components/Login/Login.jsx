<<<<<<< HEAD
import React from 'react';
import { Field, Form, Formik } from 'formik';
import { withRouter } from 'react-router';
import axios from 'axios';
import { connect } from 'react-redux';
import { compose } from 'recompose';
=======
import React from "react";
import { Field, Form, Formik } from "formik";
import { withRouter } from "react-router";
import axios from "axios";
>>>>>>> f536128973b859065490fed7f65ed039d2ab9335

import logo from "../../assets/caravan-logo-blueOnWhite.png";
import Input from "../Input/Input";
import BigButton from "../BigButton/BigButton";
import { required } from "../../utilities/formValidation";

import PersonIcon from "@material-ui/icons/PersonOutlined";
import VpnKeyIcon from "@material-ui/icons/VpnKeyOutlined";
import "./Login.scss";

export const LoginForm = () => (
	<Form>
		<Field
			icon={<PersonIcon />}
			type='text'
			tag='USERNAME'
			name='username'
			maxLength={254}
			autoFocus={true}
			validate={required}
			component={Input}
		/>
		<Field
			icon={<VpnKeyIcon />}
			type='password'
			tag='PASSWORD'
			name='password'
			autoFocus={false}
			validate={required}
			component={Input}
		/>

		<BigButton value={"Login"} />
	</Form>
);

const Login = (props) => {
<<<<<<< HEAD
	const { addLoggedUser, setLogged } = props;

=======
>>>>>>> f536128973b859065490fed7f65ed039d2ab9335
	return (
		<div className='login-page-wrapper'>
			<div className='header'>
				<img
					src={logo}
					height='25px'
					className='caravan-logo'
					alt='Caravan logo'
				/>
				<h1 className='header-text'>Welcome Back</h1>
				<p className='header-subheader'> Login to continue</p>
			</div>
			<Formik
				initialValues={{
					email: "",
					password: "",
				}}
				onSubmit={(values) => {
					const { history } = props;
					// same shape as initial values

					axios
						.post("/login", {
							username: values.username,
							password: values.password,
						})
						.then(function (response) {
							addLoggedUser(response);
							setLogged();
							history.push('/dashboard');
						})
						.catch(function (error) {
							console.log(error);
						});
				}}
			>
				{() => <LoginForm />}
			</Formik>
			<p>
				Don't have an account yet? <a href='/signup'>Sign up!</a>
			</p>
		</div>
	);
};

const mapDispatchToProps = (dispatch) => {
	return {
		addLoggedUser: (user) => {
			dispatch({ type: 'ADD_USER', payload: user });
		},
		setLogged: () => {
			dispatch({ type: 'SIGN_IN' });
		},
	};
};

const enhance = compose(withRouter, connect(null, mapDispatchToProps));

export default enhance(Login);
