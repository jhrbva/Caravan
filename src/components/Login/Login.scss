@import "../../utilities/colorPalette.scss";

.login-page-wrapper {
	margin: 15px;
	.header {
		margin-bottom: 25px;
	}
	h1 {
		color: $happy-blue;
	}
	p {
		text-align: center;
		font-size: 16px;
	}
}
