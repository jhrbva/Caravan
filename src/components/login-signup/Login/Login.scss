@import '../../../utilities/colorPalette.scss';

.login-page-wrapper {
	display: flex;
	flex-direction: column;
	align-items: center;
	justify-content: center;
}
.temp-logo {
	color: $dream-blue;
}
