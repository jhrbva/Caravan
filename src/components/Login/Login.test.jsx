import React from 'react';
import { shallow } from 'enzyme';
import Login, { LoginForm } from './Login';

describe('Login', () => {
	it('should match the snapshot', () => {
		const wrapper = shallow(<Login />);
		expect(wrapper).toMatchSnapshot();
	});
});

describe('LoginForm', () => {
	it('should have two fields', () => {
		const wrapper = shallow(<LoginForm />);
		const fieldWrapper = wrapper.find('Field');
		expect(fieldWrapper.length).toBe(2);
	});
	it('should have a username field', () => {
		const wrapper = shallow(<LoginForm />);
		const fieldWrapper = wrapper.find('Field');
		const usernameField = fieldWrapper.find({ name: 'username' });
		expect(usernameField.length).toBe(1);
	});
	it('should have a password field', () => {
		const wrapper = shallow(<LoginForm />);
		const fieldWrapper = wrapper.find('Field');
		const usernameField = fieldWrapper.find({ name: 'password' });
		expect(usernameField.length).toBe(1);
	});
});
