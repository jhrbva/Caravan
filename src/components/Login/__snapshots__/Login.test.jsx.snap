// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login should match the snapshot 1`] = `
<div
  className="login-page-wrapper"
>
  <Formik
    initialValues={
      Object {
        "email": "",
        "password": "",
      }
    }
    onSubmit={[Function]}
  >
    <Component />
  </Formik>
  <p>
    Don't have an account yet?
  </p>
  <a
    href="/signup"
  >
    Sign up!
  </a>
</div>
`;
